### PR TITLE
feat: StableMemory64: prims and tests

### PR DESCRIPTION
**Pull request by crusso**
After making internal changes to use 64 bit stablememory in #2931, this PR upgrades
* User level prims to 64 bit API
* Existing 32-bit tests to 64-bit API

### CODE REVIEW

**Comment by crusso** [comment]
@rossberg PTAL

File: test/run-drun/life-mut/life-v3.mo [hunk 1]
```diff
@@ -46,14 +46,14 @@ actor Life {
...
+    #v3 : {size : Nat; offset : Nat64}
   };
 
 
-  func ensureMemory(offset : Nat32) {
+  func ensureMemory(offset : Nat64) {
       let pagesNeeded = ((offset + 65535) / 65536) - SM.size();
       if (pagesNeeded > 0) {
-        assert (SM.grow(pagesNeeded) != 0xFFFF)
+        assert (SM.grow(pagesNeeded) != 0xFFFFFFFF)
```
Comment (rossberg) [inline test/run-drun/life-mut/life-v3.mo:56]:
```suggestion
        assert (SM.grow(pagesNeeded) != 0xFFFF_FFFF)
```

File: test/run-drun/stable-mem-nat32.mo [hunk 2]
```diff
@@ -2,16 +2,16 @@ import P "mo:⛔";
 import StableMemory "stable-mem/StableMemory";
 actor {
 
-  stable var n : Nat32 = 0;
+  stable var n : Nat64 = 0;
   assert (n == StableMemory.size());
 
-  func valOfNat32(n : Nat32) : Nat32 { n };
-  let inc : Nat32 = 4;
+  func valOfNat64(n : Nat64) : Nat32 { P.natToNat32(P.nat64ToNat(n)) };
```
Comment (rossberg) [inline test/run-drun/stable-mem-nat32.mo:8]: We really need more efficient conversions...

File: test/run-drun/stable-mem-nat64.mo [hunk 3]
```diff
@@ -2,16 +2,16 @@ import P "mo:⛔";
 import StableMemory "stable-mem/StableMemory";
 actor {
 
-  stable var n : Nat32 = 0;
+  stable var n : Nat64 = 0;
   assert (n == StableMemory.size());
 
-  func valOfNat32(n : Nat32) : Nat64 { P.natToNat64(P.nat32ToNat(n)); };
-  let inc : Nat32 = 8;
+  func valOfNat64(n : Nat64) : Nat64 { P.natToNat64(P.nat64ToNat(n)); };
```
Comment (rossberg) [inline test/run-drun/stable-mem-nat64.mo:8]: Shouldn't this be the identity now?

**Review by rossberg — APPROVED** [review]

File: test/run-drun/stable-mem-nat32.mo (hunk 2 above)
Comment (crusso) [reply test/run-drun/stable-mem-nat32.mo:8]: Yep.

File: test/run-drun/stable-mem-nat64.mo (hunk 3 above)
Comment (crusso) [reply test/run-drun/stable-mem-nat64.mo:8]:
```suggestion
  func valOfNat64(n : Nat64) : Nat64 { n };
```

**Pull request merged**